src/blog/converting-svg-to-image-png-webp-etc-in-javascript.md

```diff
@@ -0,0 +1,337 @@
+
+    const clone: SVGGraphicsElement =
+      svg.cloneNode(true) as SVGGraphicsElement;
+
+    inlineStyle({clone, style});
+
+    const base64SVG: string =
+      window.btoa(new XMLSerializer().serializeToString(clone));
+    const imgSrc: string = `data:image/svg+xml;base64,${base64SVG}`;
+
+    const image = new Image();
+
+    image.crossOrigin = 'anonymous';
+
+    image.onload = () => {
+      const canvas: HTMLCanvasElement =
+                    document.createElement('canvas');
+
+      canvas.width = width;
+      canvas.height = height;
+
+      const context: CanvasRenderingContext2D | null =
+        canvas.getContext('2d');
+      context?.drawImage(image, 0, 0, width, height);
+
+      resolve(canvas);
+    };
+
+    image.src = imgSrc;
+  });
+};
+```
+
+As I modified its declaration, I also had to change the caller in order to find and path the style of the text element.
+
+```javascript
+@Method()
+async toBlob(type: string = 'image/webp'): Promise<Blob> {
+  const style: CSSStyleDeclaration | undefined =
+  this.textRef ? getComputedStyle(this.textRef) : undefined;
+
+  const canvas: HTMLCanvasElement =
+    await svgToCanvas({svg: this.svgRef, style});
+  return canvasToBlob({canvas, type});
+}
+```
+
+*****
+
+#### Canvas To Image (Blob)
+
+Converting the canvas to an image result in a blob. In my original solution, I implemented such transformation with the help of the fetch API. It’s clean and compact.
+
+```javascript
+export const canvasToBlob =
+  async ({canvas, type}: {canvas: HTMLCanvasElement; type: string}):
+    Promise<Blob> => {
+    const dataUrl: string = canvas.toDataURL(type);
+    return (await fetch(dataUrl)).blob();
+  };
+```
+
+However, once again you might say 😅, I discovered an issue at runtime when I deployed my application.
+
+Such approach requires enabling `data:` in the `connect-src` rule of a content security policy (CSP) which is strongly discouraged.
+
+Fortunately, there is another way to convert canvas to blob, the built-in [toBlob()](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/toBlob) method that accept a `callback` as argument.
+
+```javascript
+export const canvasToBlob =
+  async ({canvas, type}: {canvas: HTMLCanvasElement; type: string}):
+    Promise<Blob> => {
+    return new Promise<Blob>((resolve) => canvas.toBlob((blob: Blob) => resolve(blob), type));
+  };
+```
+
+*****
+
+### Conclusion
+
+Sometimes development takes a bit more time than excepted, it took me a day to develop, solve all issues, test and publish the component but, I am glad I did it.
+
+Not only it resolves a feature I needed to notably publish this blog post but, I learned quite some new tricks along the way.
+
+Merry Christmas 🎄
+David
+
+*****
+
+### Further Reading
+
+Wanna read more our project ?We are porting [DeckDeckGo](https://deckdeckgo.com/) to [DFINITY’s](https://dfinity.org/) Internet Computer. Here is the list of blog posts I published since we started the project:
```

Comment:
```suggestion
Wanna read more about our project? We are porting [DeckDeckGo](https://deckdeckgo.com/) to [DFINITY’s](https://dfinity.org/) Internet Computer. Here is the list of blog posts I published since we started the project:
```